src/services/photos/utils/photosUtils.ts

```diff
@@ -121,13 +121,23 @@ export class PhotosUtils {
     return hash.digest();
   }
 
-  public getPhotoPath({ name, size, type }: { name: string; size: PhotoSizeType; type: string }) {
+  public getPhotoPath({
+    name,
+    size,
+    takenAt,
+    type,
+  }: {
+    name: string;
+    takenAt: number;
+    size: PhotoSizeType;
+    type: string;
+  }) {
     if (size === PhotoSizeType.Full) {
-      return `${PHOTOS_FULL_SIZE_DIRECTORY}/${name}${type === 'unknown' || !type ? '' : '.' + type}`;
+      return `${PHOTOS_FULL_SIZE_DIRECTORY}/${name}_${takenAt}${type === 'unknown' || !type ? '' : '.' + type}`;
```

Comment:
I suggesting using something that is not user-modifiable as a UUID mapped to the local database, just to ensure that nor the date neither name changing gives problems

Reply:
Just a suggestion (not a part of the required changes)

Reply:
the problem with that is that you will need then to maintain the relationship between db and filesystem, some things that are hard to do here:

- Detect why and when a file was removed, then you will need to update the db

The paths are made this way so we can quickly found them in the filesystem, if we use the UUID, we will need first to look in the db, then in the filesystem, if the item is not there, update the db and update the filesystem, which are many bridge trips just for this.

According to how Photos works, if a user renames a photo, or changes the takenAt, we understand is a new photo, the hash that makes that photo unique will change too, as that won't match the backed up photo.